app/src/main/java/com/github/gotify/messages/MessagesActivity.kt

```diff
@@ -0,0 +1,633 @@
+        } catch (e: IOException) {
+            Log.e("Problem evicting Picasso cache", e)
+        }
+        startActivity(Intent(this, InitializationActivity::class.java))
+        finish()
+    }
+
+    private fun onRefresh() {
+        viewModel.messages.clear()
+        launchCoroutine {
+            loadMore(viewModel.appId)
+        }
+    }
+
+    private fun openDocumentation() {
+        val browserIntent = Intent(Intent.ACTION_VIEW, Uri.parse("https://gotify.net/docs/pushmsg"))
+        startActivity(browserIntent)
+    }
+
+    private fun onUpdateApps(applications: List<Application>) {
+        val menu: Menu = binding.navView.menu
+        menu.removeGroup(R.id.apps)
+        viewModel.targetReferences.clear()
+        updateMessagesAndStopLoading(viewModel.messages[viewModel.appId])
+        var selectedItem = menu.findItem(R.id.nav_all_messages)
+        applications.indices.forEach {
+            val app = applications[it]
+            val item = menu.add(R.id.apps, it, APPLICATION_ORDER, app.name)
+            item.isCheckable = true
+            if (app.id == viewModel.appId) selectedItem = item
+            val t = Utils.toDrawable(resources) { icon: Drawable? ->
+                item.icon = icon
+            }
+            viewModel.targetReferences.add(t)
+            viewModel.picassoHandler
+                .get()
+                .load(Utils.resolveAbsoluteUrl(viewModel.settings.url + "/", app.image))
+                .error(R.drawable.ic_alarm)
+                .placeholder(R.drawable.ic_placeholder)
+                .resize(100, 100)
+                .into(t)
+        }
+        selectAppInMenu(selectedItem)
+    }
+
+    private fun initDrawer() {
+        setSupportActionBar(binding.appBarDrawer.toolbar)
+        binding.navView.itemIconTintList = null
+        val toggle = ActionBarDrawerToggle(
+            this,
+            binding.drawerLayout,
+            binding.appBarDrawer.toolbar,
+            R.string.navigation_drawer_open,
+            R.string.navigation_drawer_close
+        )
+        binding.drawerLayout.addDrawerListener(toggle)
+        toggle.syncState()
+        binding.navView.setNavigationItemSelectedListener(this)
+        val headerView: View = binding.navView.getHeaderView(0)
+        val settings = viewModel.settings
+        val user = headerView.findViewById<TextView>(R.id.header_user)
+        user.text = settings.user?.name
+        val connection = headerView.findViewById<TextView>(R.id.header_connection)
+        connection.text = getString(R.string.connection, settings.user?.name, settings.url)
+        val version = headerView.findViewById<TextView>(R.id.header_version)
+        version.text =
+            getString(R.string.versions, BuildConfig.VERSION_NAME, settings.serverVersion)
+        val refreshAll = headerView.findViewById<ImageButton>(R.id.refresh_all)
+        refreshAll.setOnClickListener {
+            onRefreshAll(it)
+        }
```

Comment:
Could you add some empty lines to the methods in this file, it was pretty difficult to read? It could be similar to the old java files.

Reply:
Sorry.
Resolved in 228b7fbba017aec3b26f1f89aff1aca138d6f6f6